contracts/hermez/lib/InstantWithdrawManager.sol

```diff
@@ -247,19 +219,21 @@ contract InstantWithdrawManager is HermezHelpers {
         uint256 amountUSD = _token2USD(tokenAddress, amount);
         if (amountUSD == 0) return true;
 
-        uint256 bucketIdx = _findBucketIdx(amountUSD);
+        int256 bucketIdx = _findBucketIdx(amountUSD);
+        if (bucketIdx == -1) return true;
 
-        Bucket storage currentBucket = buckets[bucketIdx];
-        if (currentBucket.withdrawals > 0) {
-            return true;
-        } else {
-            uint256 differenceBlocks = block.number.sub(
-                currentBucket.blockStamp
-            );
-            if (differenceBlocks < currentBucket.blockWithdrawalRate)
-                return false;
-            else return true;
-        }
+
+        (, uint256 blockStamp, uint256 withdrawals, uint256 rateBlocks, uint256 rateWithdrawals, uint256 maxWithdrawals) = unpackBucket(buckets[bucketIdx]);
+
+        uint256 differenceBlocks = block.number.sub(blockStamp);
+        uint256 periods = differenceBlocks.div(rateBlocks);
+
+        withdrawals = withdrawals.add( periods.mul(rateWithdrawals));
+        if (withdrawals>maxWithdrawals) withdrawals = maxWithdrawals;
```

Comment:
this function does not change the state, so, it's not necessary to calculate this